elastic/apm-server
# processor/otel: set outcome from status code

## Motivation/summary

For Jaeger HTTP transactions/spans, set outcome to success or failure.
For transactions we set to success for codes < 500, and for spans we set
to success for codes < 400, in line with our own agents.

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

Instrument a web service that receives and sends HTTP requests with Jaeger. Send spans to APM Server, observe that `event.outcome` is added to the transactions and spans according to the HTTP status codes.

## Related issues

Closes #4117
